sdl_android/src/main/java/com/smartdevicelink/api/SdlManager.java

```diff
@@ -503,20 +506,19 @@ public void sendRPCs(List<? extends RPCMessage> rpcs, final OnMultipleRequestLis
 
 	/**
 	 * Starts up a SdlManager, and calls provided callback called once all BaseSubManagers are done setting up
-	 * @param listener CompletionListener that is called once the SdlManager state transitions
-	 * from SETTING_UP to READY or ERROR
+	 * @param listener ManagerListener that is called when the SdlManager is ready / failed to start, or is destroyed
 	 */
 	@SuppressWarnings("unchecked")
-	public void start(@NonNull CompletionListener listener){
-		initListener = listener;
+	public void start(@NonNull SdlManagerListener listener){
```

Comment:
Since this listener is more than a simple completion listener should we add this to the builder instead?

Reply:
We could additionally have `onReady` and `onStart`

Reply:
I understand the original purpose of putting this listener in the `start` method, but we have expanded the role of the listener so it no longer makes sense to be added during the singular process of start. The developer does not call `proxy.start()` so there is no difference between started and ready in the listeners context, the developer should only be notified when the manager can be used.  The `onStart` naming would be to follow closer to Android callbacks (Activity, Service, etc), but I'm fine with `onReady` instead.